docs/functionalabstractions/index.md

```diff
@@ -3,4 +3,289 @@ id: index
+    new Associative[Int] {
+      def combine(left: Int, right: Int): Int =
+        left + right
+    }
+
+  implicit def ListAssociative[A]: Associative[List[A]] =
+    new Associative[List[A]] {
+      def combine(left: List[A], right: List[A]): List[A] =
+        left ::: right
+    }
+}
+```
+
+If the instance of the type class depends on other parameters, like the `A` in `ListAssociative` we define it as an `implicit def`. Otherwise, we define it as an `implicit val`.
+
+We can think of the `implicit` keyword as associating the type `Int` with the value `IntAssociative`. So now if we ask the Scala compiler for the `Associative` instance for `Int` it will be able to find it.
+
+In the type class pattern we also typically define extension methods that will be available on any data type for which an instance of the type class is defined.
+
+```scala mdoc
+implicit final class AssociativeSyntax[A](private val self: A) {
+  def <>(that: A)(implicit associative: Associative[A]): A =
+    associative.combine(self, that)
+}
+```
+
+This machinery allows us to use the `<>` operator to combine values of any type as long as an `Associative` instance is defined for it.
+
+```scala mdoc
+val int: Int =
+  1 <> 2
+
+val list: List[Int] =
+  List(1, 2, 3) <> List(4, 5, 6)
+```
+
+Of course we didn't really need all of this machinery to add two numbers or concatenate two lists, but where this pattern gets powerful is when we can use it to combine more complex data types in a principled way.
+
+For example, say we have an application where users can vote on content they are interested in learning more about. We might have a data structure to keep track of the number of votes for different topics like this.
+
+```scala mdoc
+final case class Topic(value: String)
+final case class Votes(value: Int)
+final case class VoteMap(map: Map[Topic, Votes])
+```
+
+A common thing we might want to do is combine two `VoteMap` values, for example if the user has a local copy of the `VoteMap` and we want to update it with a new batch of votes from the server.
+
+We could do that manually like this.
+
+```scala mdoc:nest
+final case class Topic(value: String)
+
+final case class Votes(value: Int) { self =>
+  def combine(that: Votes): Votes =
+    Votes(self.value + that.value)
+}
+
+final case class VoteMap(map: Map[Topic, Votes]) { self =>
+  def combine(that: VoteMap): VoteMap =
+    VoteMap(that.map.foldLeft(self.map) { case (map, (topic, votes)) =>
+      map + (topic -> map.getOrElse(topic, Votes(0)).combine(votes))
+    })
+}
+```
+
+This isn't the worst but it isn't really the kind of code we want to be writing. We want to be thinking about the logic of our application rather than how to combine maps.
+
+This is where ZIO Prelude can help.
+
+The way we're combining these maps actually follows a pattern. If a value is in either map we include it in the combined map and if a value appears in both maps we combine the keys.
```

Comment:
I understand it differently. See what you think of my wording, which I guess you'll reject?

If a key is in a single map we include it in the combined map with its associated value and if a key appears in both maps we combine the associated values of the maps for that key.

Reply:
No I think that is good. Changed along those lines.